MaintenanceDashboard.Data/API/ReceivedPaddleContext.cs

```diff
@@ -28,99 +23,49 @@ public void CreateReceivedPaddle(ReceivedPaddle receivedPaddle)
             CheckValue.RequireString(receivedPaddle.PlannedRepairDate);
             CheckValue.RequireString(receivedPaddle.IsOrders);
             CheckValue.RequireString(receivedPaddle.ReceivingEmployee);
-            //CheckValue.RequireString(receivedPaddle.PaddleNumber);
 
-            context.ReceivedPaddles.Add(receivedPaddle);
-            context.SaveChanges();
+            _context.ReceivedPaddles.Add(receivedPaddle);
+            _context.SaveChanges();
         }
 
         public Employee CheckEmployee(Employee employee)
         {
-            return context.Employees.FirstOrDefault(c => c.Id == employee.Id);
+            return _context.Employees.FirstOrDefault(c => c.Id == employee.Id);
         }
 
         public void CreateSpendedPaddle(SpendedPaddle spendedPaddle)
         {
-            context.SpendedPaddles.Add(spendedPaddle);
-            context.SaveChanges();
+            _context.SpendedPaddles.Add(spendedPaddle);
+            _context.SaveChanges();
         }
 
         public void DeleteReceivedPaddle(ReceivedPaddle receivedPaddle)
         {
-            context.ReceivedPaddles.Remove(receivedPaddle);
-            context.SaveChanges();
+            _context.ReceivedPaddles.Remove(receivedPaddle);
+            _context.SaveChanges();
         }
 
         public void UpdateLastPreventionDate(ReceivedPaddle receivedPaddle)
         {
             if (receivedPaddle.ActivityPerformed == "Prewencja")
             {
                 var t =
-                    (from c in context.Paddles
+                    (from c in _context.Paddles
                      where c.Id == receivedPaddle.PaddleId
                      select c).First();
                 t.LastPrevention = DateTime.Now.ToString("MM/dd/yyyy");
 
-                context.SaveChanges();
+                _context.SaveChanges();
             }
 
         }
 
-        public int CheckForeignKey(string numer)
-        {
-            return context.Paddles.FirstOrDefault(c => c.Number == numer).Id;
-        }
-
-        public bool CheckReceivedPaddleExist(string number)
-        {
-            var result = context.Paddles.FirstOrDefault(c => c.Number == number);
-
-            if (result == null)
-                return true;
-
-            return false;
-        }
-
-        public bool CheckIfIsAccepted(string number)
-        {
-            var result = context.ReceivedPaddles.FirstOrDefault(c => c.Paddle.Number == number);
+        public int CheckForeignKey(string number) => _context.Paddles.FirstOrDefault(c => c.Number == number)?.Id ?? 0;
```

Comment:
`FirstOrDefault` zwraca: 
- pierwszy element jeśli znajdzie kilka
- nulla jeśli nie znajdzie pasującego elementu

zapis `...?.Id` mówi nam o tym że `jeśli znajdę Paddle to pobierz jego Id. Jeśli FirstOrDefault zwróci null to nic nie rób`.  To co napiszesz po znaku zapytania wykona się tylko wtedy, kiedy po lewej stronie nie jest null.

Operator `??` mówi nam o tym, że jeśli po lewej stronie wyrażenia jest null, to zrób to co jest po prawej. W tym przypadku jeśli nie znaleziono `Paddle`, wynik to `null` to zwróć zero. Dlaczego tak to napisałem: masz wyjaśnienie wyżej. 

Poza tym przemyśl ten kod, na pierwszy rzut oka coś mi tu nie gra. Jeśli `Number` jest unikalne to okej, ale jeśli jest unikalne to dlaczego nie użyjesz `SingleOrDefault`? Wtedy mówisz programiście "Hej, tutaj jest dokładnie jedna encja o takim polu NUmber w bazie i nie może być drugiej takiej". A tak, ja jako drugi programista nie do końca wiem co się ma zadziać, kiedy będą trzy encje o tym samym `Number`, Z Tego kodu nie wynika, z której encji `Paddle` dostane Id

Reply:
Poza tym, pod to koniecznie przydałby się test jednostkowy by pokazać, że działa to tak jak chcesz (patrz moje wątpliwości między `First` a `Single` or default). Więcej o testach w .necie:

https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices